### PR TITLE
nginx + puma

### PR DESCRIPTION
**Pull request by RDrop**
I cleaned up my code to only add nginx + puma to your most recent push.
It installs clean with the server accessible and running.
Also added is a redis upstart conf that can be used to start-stop the daemon.

### CODE REVIEW

**Comment by RDrop** [comment]
...I didn't change the virtual_boxes_spec.rb file

File: devops/group_vars/all/config.yml [hunk 1]
```diff
@@ -43,3 +43,9 @@ unicorn_config_path: "{{ app_config_path }}/unicorn.rb"
 unicorn_pid_path: "{{ app_temp_path }}/unicorn.pid"
 unicorn_stderr_path: "{{ app_logs_path }}/unicorn.err.log"
 unicorn_stdout_path: "{{ app_logs_path }}/unicorn.log"
+
+puma_bind_path: "unix://{{ app_temp_path }}/sockets/puma.sock"
+puma_config_path: "{{ app_config_path }}/puma.rb"
+puma_pidfile_path: "{{ app_temp_path }}/pids/puma.pid"
+puma_stderr_redirect_path: "{{ app_logs_path }}/puma.err.log"
+puma_stdout_redirect_path: "{{ app_logs_path }}/puma.log"
```
Comment (andreychernih) [inline devops/group_vars/all/config.yml]: @RDrop everything under `devops` is only used for `railsbox.io` deployment (which uses unicorn), so it's better to revert this

File: templates/ansible/roles/nginx/templates/nginx_puma.conf.j2 [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  listen 80;
+  listen 443 ssl;
+
+  sendfile off;
+
+  server_name {{ server_name }};
+
+  ssl_certificate {{ ssl_certificate_path }};
+  ssl_certificate_key {{ ssl_certificate_key_path }};
+  ssl_protocols  TLSv1 TLSv1.1 TLSv1.2;  # don’t use SSLv3 ref: POODLE
```
Comment (andreychernih) [inline templates/ansible/roles/nginx/templates/nginx_puma.conf.j2]: @RDrop I am not sure if SSL is supported currently, let's stick with http for now

File: templates/ansible/roles/nginx/templates/nginx_puma.conf.j2 [hunk 3]
```diff
@@ -0,0 +1,34 @@
+# {{ ansible_managed }}
+
+upstream app {
+  server {{ puma_bind_path }} fail_timeout=0;
+}
+
+server {
+  listen 80;
+  listen 443 ssl;
```
Comment (andreychernih) [inline templates/ansible/roles/nginx/templates/nginx_puma.conf.j2]: ditto

File: templates/ansible/roles/redis/tasks/main.yml [hunk 4]
```diff
@@ -14,3 +14,7 @@
 - name: Configure redis-rb
   template: src=redis.rb.j2 dest={{ app_config_path }}/initializers/redis.rb owner={{ user_name }} backup=yes
   when: redis_orm is defined and redis_orm == 'redis-rb'
+ 
+- name: Create upstart config
+  template: src=upstart.conf.j2 dest=/etc/init/redis.conf
+  notify: restart redis
```
Comment (andreychernih) [inline templates/ansible/roles/redis/tasks/main.yml]: I believe `redis-server` Ubuntu package should create `init.d` script for starting redis. Will it conflict with upstart?

**Comment by andreychernih** [comment]
Thanks for extracting this. LGTM in general, please take a look into [failing test](https://travis-ci.org/andreychernih/railsbox/builds/55267139) and a few comments.

File: templates/ansible/roles/nginx/templates/nginx_puma.conf.j2 (hunk 2 above)
Comment (andreychernih) [inline templates/ansible/roles/nginx/templates/nginx_puma.conf.j2]: Nevermind, I totally forgot that I've added SSL as well :) Please ignore this comment.

**Comment by andreychernih** [comment]
I needed your changes so I went ahead and merged this in. Thanks for the contribution!

**Pull request merged**